Added doc section 'Configuring the robot task' (#2536)
SFDO-Tooling/CumulusCI: docs/robot.rst

  documentation if you choose to create a library of robot keywords
  for your project.

> Configuring the robot task

Reviewer: ```suggestion
Configuring the ``robot`` task
```

--------------------------

> The robot framework command-line test runner supports more than 50

Reviewer: ```suggestion
The Robot Framework command-line test runner supports more than 50
```

`command line options
<http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#command-line-options-for-test-execution>`_.

> To make the robot task simpler to use, we've only exposed a few of the

Reviewer: ```suggestion
To make the ``robot`` task simpler to use, we've only exposed a few of the
```

> command-line options at the task level. For example, the robot task

Reviewer: ```suggestion
command-line options at the task level. For example, the Robot task
```

> options ``include`` and ``exclude`` directly map to the robot

Reviewer: ```suggestion
options ``include`` and ``exclude`` directly map to the ``robot``
```

command-line options ``--include`` and ``--exclude``. These options

> can be configured the same as any other task option. Other robot

Reviewer: ```suggestion
are specified the same way as task options elsewhere in the CumulusCI framework, 
using either command-line options as shown above or by including them in the ``options``
section of a task configuration in ``cumulusci.yml``::

    tasks:
        robot:
            options:
                include: <value>

Other Robot
```

command-line options, such as ``tagstatlink``, ``expandkeywords``, and
many others, have no direct task option counterpart.

> There may be times when you want to use some of the robot framework

Reviewer: ```suggestion
There may be times when you want to use some of the Robot Framework
```

options which haven't been exposed as task options. We support that through

> an additional ``options`` section inside the normal task options in

Reviewer: ```suggestion
an additional ``options`` section nested inside the typical task options in
```

> cumulusci.yml.

Reviewer: ```suggestion
``cumulusci.yml``.
```

For example, one of the most common uses of this inner ``options`` section is to

> use the robot command line option ``--outputdir`` to specify where robot should

Reviewer: ```suggestion
use the Robot command line option ``--outputdir`` to specify where Robot should
```

> write its report and log files. To configure this command line option

Reviewer: ```suggestion
write its report and log files. To configure this option
```

for the task, you must remove the leading dashes from the option name
and then place that option

> and value in a nested ``option`` section.

Reviewer: ```suggestion
and value in a nested ``options`` section.
```

.. code-block:: robotframework

    tasks:
        robot:
            options:
                options:
                    outputdir: robot/my_project/results

> Any robot command-line option which takes a value can be specified

Reviewer: ```suggestion
Any Robot command-line option which takes a value can be specified
```

Author: I struggle with this sort of thing all the time. I get capitalizing `Robot Framework`, and `Robot` is a reasonable abbreviation. But `robot` is the actual name of the tool so that makes sense too.

Reviewer: What do you think about referring to the underlying Robot tool as "Robot CLI", to avoid confusion with our `robot` task?

this way. For example, to use the robot option ``--name`` along with

> ``--outputdir``, your cumulusci.yml file should look like

Reviewer: ```suggestion
``--outputdir``, your ``cumulusci.yml`` file should look like
```

this:

.. code-block:: robotframework

    tasks:
        robot:
            options:
                options:
                    outputdir: robot/my_project/results
                    name: Salesforce Robot Tests

> Configuring the libdoc task

Reviewer: ```suggestion
Configuring the ``libdoc`` task
```
